coffee/web_socket_channel.coffee

```diff
@@ -30,18 +31,37 @@ class palava.WebSocketChannel extends @EventEmitter
       @emit 'open', handshake
     @socket.onmessage = (msg) =>
       try
-        @emit 'message', JSON.parse(msg.data)
+        parsedMsg = JSON.parse(msg.data)
+        if parsedMsg == {event: "pong"}
+          @outstandingPongs -= 1
```

Comment:
I would probably still reset to 0 as any answer confirms that we have a connection. But that's just a decision between detecting super slow connections and literally dead connections.